src/ui-bootstrap/src/types/select.ts

```diff
@@ -56,26 +58,31 @@ export class FormlyFieldSelect extends FieldType {
   get valueProp(): string { return this.to.valueProp || 'value'; }
   get groupProp(): string { return this.to.groupProp || 'group'; }
 
-  get selectOptions() {
-    const options: SelectOption[] = [],
-      groups: { [key: string]: SelectOption[] } = {};
+  get selectOptions(): Observable<any[]> {
+    if (!(this.to.options instanceof Observable)) {
+      const options: SelectOption[] = [],
+        groups: { [key: string]: SelectOption[] } = {};
 
-    this.to.options.map((option: SelectOption) => {
-      if (!option[this.groupProp]) {
-        options.push(option);
-      } else {
-        if (groups[option[this.groupProp]]) {
-          groups[option[this.groupProp]].push(option);
+      this.to.options.map((option: SelectOption) => {
+        if (!option[this.groupProp]) {
+          options.push(option);
         } else {
-          groups[option[this.groupProp]] = [option];
-          options.push({
-            label: option[this.groupProp],
-            group: groups[option[this.groupProp]],
-          });
+          if (groups[option[this.groupProp]]) {
+            groups[option[this.groupProp]].push(option);
+          } else {
+            groups[option[this.groupProp]] = [option];
+            options.push({
+              label: option[this.groupProp],
+              group: groups[option[this.groupProp]],
+            });
+          }
         }
-      }
-    });
+      });
 
-    return options;
+      return of(options);
+    } else {
+      // return observable directly
+      return this.to.options || of([]);
```

Comment:
` || of([])` can be removed

Reply:
we should apply a `map` operator to apply `groupProp`

Reply:
> we should apply a map operator to apply groupProp

Can you elaborate on this?

Reply:
we should add support for grouped select when passing an observable

Reply:
Oh, sorry 😅, didn't understand at first. Yep, true. I'll open a separate issue for that.